### PR TITLE
feat(form): support a generic control container

### PR DESCRIPTION
**Pull request by gnomeontherun**
This implements a new `clr-control-container` component and `clrControl` directive for wrapping potentially any form control to work with Clarity forms. Any controls must implement FormControlAccessor to properly wire up with the forms logic, but it works for internal or third party components.

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [x] Feature
* [ ] Code style update (formatting, local variables)
* [x] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

closes: #2886 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/dev/src/app/forms/generic-container/generic-container.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@import '';
```
Comment (hippee-lee) [inline src/dev/src/app/forms/generic-container/generic-container.scss]: nit: empty import or empty file?

File: src/website/src/app/documentation/demos/forms/ng/generic.html [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<!-- Template Driven Example -->
+<form clrForm>
+  <input clrControl placeholder="Basic text" name="basic" [(ngModel)]="model.basic" />
```
Comment (hippee-lee) [inline src/website/src/app/documentation/demos/forms/ng/generic.html]: I thought we wanted to only show examples with a label (same comment for lines 6, 34, & 37) b/c thats the guidance we give for forms.

File: src/clr-angular/forms/checkbox/checkbox-container.ts [hunk 3]
```diff
@@ -4,15 +4,12 @@
...
-import { Subscription } from 'rxjs';
-import { NgControl } from '@angular/forms';
+import { Component, Input } from '@angular/core';
 
 import { IfErrorService } from '../common/if-error/if-error.service';
-import { ClrLabel } from '../common/label';
 import { ControlClassService } from '../common/providers/control-class.service';
-import { LayoutService } from '../common/providers/layout.service';
 import { NgControlService } from '../common/providers/ng-control.service';
+import { ClrAbstractContainer } from '../common/abstract-container';
```
Comment (mathisscott) [inline src/clr-angular/forms/checkbox/checkbox-container.ts]: I love how this effort cleaned up so much repeated code.

File: src/clr-angular/forms/common/abstract-container.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      })
+    );
+  }
+
+  controlClass() {
+    return this.controlClassService.controlClass(this.invalid, this.addGrid());
+  }
+
+  addGrid() {
+    return this.layoutService && !this.layoutService.isVertical();
```
Comment (mathisscott) [inline src/clr-angular/forms/common/abstract-container.ts]: nit: For a bit of fun, you should really try Typescript's optional chaining...

```
return this.layoutService?.isVertical()
```

Comment (gnomeontherun) [reply src/clr-angular/forms/common/abstract-container.ts]: Something in our compilation process doesn't like this yet :(

File: src/website/src/app/documentation/demos/forms/ng/generic.html (hunk 2 above)
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/forms/ng/generic.html]: Good point, I'll remove the line.

**Comment by rbirkgit** [comment]
Any chance this can be back ported to 2.x?

**Pull request merged**